src/Sentry.AzureFunctions.Worker/SentryFunctionsWorkerMiddleware.cs

```diff
@@ -0,0 +1,82 @@
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Middleware;
+
+namespace Sentry.AzureFunctions.Worker;
+
+internal class SentryFunctionsWorkerMiddleware : IFunctionsWorkerMiddleware
+{
+    private readonly IHub _hub;
+
+    public SentryFunctionsWorkerMiddleware(IHub hub)
+    {
+        _hub = hub;
+    }
+
+    public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next)
+    {
+        try
+        {
+            var transactionName = await GetTransactionNameAsync(context) ?? context.FunctionDefinition.Name;
+
+            _hub.ConfigureScope(scope =>
+            {
+                var transaction = _hub.StartTransaction(transactionName, "function");
+                scope.Transaction = transaction;
+
+                scope.UnsetTag("AzureFunctions_FunctionName");
+                scope.UnsetTag("AzureFunctions_InvocationId");
```

Comment:
A comment here would be useful, describing how these were set in `SentryLogger` because they were passed in as `state` when the functions runtime was logging.

Or maybe a better way to handle this would be to prevent them from being added in the first place?  We could make an option on `SentryLoggingOptions` for tag filters, and then use them in the `SentryLogger`.  Then our Azure Functions library could just set these as tag filters during initialization.

Reply:
> Or maybe a better way to handle this would be to prevent them from being added in the first place?  We could make an option on `SentryLoggingOptions` for tag filters, and then use them in the `SentryLogger`.  Then our Azure Functions library could just set these as tag filters during initialization.

I actually like this idea a lot more. The concert for those tags shouldn't be with the Functions middleware. Plus, it could benefit other Sentry integrations that would like to filter out other tags, right?

So I'm assuming the actual filtering would happen in `SentryLogger` [here](https://github.com/getsentry/sentry-dotnet/blob/0110385baedc3527d03652b4b6a9dbd8fe459029/src/Sentry.Extensions.Logging/SentryLogger.cs#L104). Should this be part of this PR or do you think it should be a follow-up PR? I'm inclined to include it in this work.

Reply:
It would be further up, [here](https://github.com/getsentry/sentry-dotnet/blob/0110385baedc3527d03652b4b6a9dbd8fe459029/src/Sentry.Extensions.Logging/SentryLogger.cs#LL91C43-L91C43) based on the property key (which becomes the tag name).

It's small enough, but since it's not specific to Functions, it should be in it's own PR so we can call it out as a separate feature.  You can either do it first and merge it in for usage here, or you can leave this one with the current approach and fix it when you do that later.

Reply:
Note to self - remove `UnsetTags` and validate it's working.

Reply:
The necessary change will be completed in #2360